sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileStatusCacheSensorTest.java

```diff
@@ -0,0 +1,146 @@
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.cache.WriteCache;
+import org.sonar.api.batch.sensor.internal.DefaultSensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileStatusCacheSensorTest {
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test
+  public void should_describe() {
+    var sensorDescriptor = new DefaultSensorDescriptor();
+    var sensor = new FileStatusCacheSensor(new HashProvider());
+    sensor.describe(sensorDescriptor);
+
+    assertThat(sensorDescriptor.name()).isEqualTo("File status cache sensor");
+    assertThat(sensorDescriptor.languages()).isEmpty();
+  }
+
+  @Test
+  public void execute_whenAnalyzingAPullRequest_logsMessage() throws IOException {
+    var settings = new MapSettings().setProperty("sonar.pullrequest.key", "42");
+    var context = SensorContextTester.create(temp.newFolder()).setSettings(settings);
+    var sensor = new FileStatusCacheSensor(new HashProvider());
+
+    sensor.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Incremental PR analysis: Cache is not updated for pull requests.");
+  }
+
+  @Test
+  public void execute_whenPullRequestCacheBasePathIsNotConfigured_logsWarning() throws IOException {
+    var context = SensorContextTester.create(temp.newFolder());
+    var sut = new FileStatusCacheSensor(new HashProvider());
+
+    sut.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Incremental PR analysis: Pull request cache base path is not configured. Skipping file status upload.");
+  }
+
+  @Test
+  public void execute_whenCacheIsDisabled_logsWarning() throws IOException {
+    var settings = new MapSettings().setProperty("sonar.pullrequest.cache.basepath", "C:/");
+    var context = SensorContextTester.create(temp.newFolder()).setSettings(settings);
+    context.setCacheEnabled(false);
+    var sensor = new FileStatusCacheSensor(new HashProvider());
+
+    sensor.execute(context);
+
+    assertThat(logTester.logs()).containsExactly("Incremental PR analysis: Analysis cache is disabled. Skipping file status upload for incremental PR analysis.");
+  }
+
+  @Test
+  public void execute_whenCacheIsEnabled_itAddsTheFiles() throws IOException, NoSuchAlgorithmException {
+    var hashProvider = mock(HashProvider.class);
+    when(hashProvider.computeHash(any())).thenReturn(new byte[0]);
```

Comment:
Let's return `{42,42}` as a hash. And assert that this value was pushed to cache
```suggestion
    when(hashProvider.computeHash(any())).thenReturn(new byte[0]);
```